suplemon/viewer.py

```diff
@@ -50,76 +51,38 @@ def __init__(self, app, window):
         self.x_scroll = 0
         self.cursors = [Cursor()]
 
-        # Lexer for translating tokens to strings
-        self.lexer = None
-        # Built in syntax definition (for commenting etc.)
-        self.syntax = None
-        # Normal Pygments lexer
-        self.pygments_syntax = None
+        # Copy/paste buffer
+        self.buffer = []
+
+        # Runnable methods
+        self.operations = {
+            "arrow_right": self.arrow_right,              # Arrow Right
+            "arrow_left": self.arrow_left,                # Arrow Left
+            "arrow_up": self.arrow_up,                    # Arrow Up
+            "arrow_down": self.arrow_down,                # Arrow Down
+            "jump_left": self.jump_left,                  # Ctrl + Left
+            "jump_right": self.jump_right,                # Ctrl + Right
+            "jump_up": self.jump_up,                      # Ctrl + Up
+            "jump_down": self.jump_down,                  # Ctrl + Down
+            "page_up": self.page_up,                      # Page Up
+            "page_down": self.page_down,                  # Page Down
+            "home": self.home,                            # Home
+            "end": self.end,                              # End
+            "find": self.find_query                       # Ctrl + F
+        }
```

Comment:
I feel that the decorator might be a good choice for now although I'm ok with the current solution too. However in some cases the operation name might be different from the method name (like `find`). The current mapping is used also for the <kbd>Ctrl</kbd>+<kbd>E</kbd> run command prompt. I think it's handy to have the operation names stored as strings.

How would you incorporate the decorator into the class? I haven't used decorators much. This is related but didn't really solve my problem (specifically adding operations on `__init__`)  http://stackoverflow.com/questions/1263451/python-decorators-in-classes ?